unit-test-security-rules-v9/test/database.spec.mjs

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import assert from 'assert';
+import { readFileSync, createWriteStream } from 'fs';
+import http from 'http';
+
+// Workarounds for npm not recognizing named exports in cjs.
+import appNs from 'firebase/app';
+const { initializeApp, deleteApp, getApps, setLogLevel } = appNs;
+import dbNs from 'firebase/database';
+const { getDatabase, useDatabaseEmulator, ref, get, update } = dbNs;
+
+import adminAppNs from 'firebase-admin/app';
+const { initializeApp: initializeAdminApp } = adminAppNs;
+import adminDatabaseNs from 'firebase-admin/database';
+const { getDatabase: getAdminDatabase } = adminDatabaseNs;
+
+/**
+ * The emulator will accept any database name for testing.
+ */
+const DATABASE_NAME = "demo-database-emulator";
+
+/**
+ * The FIREBASE_DATABASE_EMULATOR_HOST environment variable is set automatically
+ * by "firebase emulators:exec"
+ */
+const { FIREBASE_DATABASE_EMULATOR_HOST } = process.env;
```

Comment:
I guess all this is to avoid importing RUT and running `discoverEmulators()`?

Reply:
TBH this is probably more straightforward to set up, since the env var is also needed to configure `firebase-admin`. We'd better ensure it is set or it may hit prod.